src/t/d17.clj

```diff
@@ -0,0 +1,258 @@
+                   )]
+    [(parse-long A) (parse-long B) (parse-long C) instr]))
+
+
+(defn combo-oprand
+  [A B C oprand]
+  (cond
+    (<= oprand 3) oprand
+    (= 4 oprand) A
+    (= 5 oprand) B
+    (= 6 oprand) C
+    "7 or none of the above is pbm" (prn "Problem with oprand" oprand)))
+
+(defn adv
+  [A B C oprand p]
+  [(quot A (int (math/pow 2 (combo-oprand A B C oprand))))
+   B
+   C
+   nil
+   (+ 2 p)])
+
+(defn bxl
+  [A B C oprand p]
+  [A
+   (bit-xor B oprand)
+   C
+   nil
+   (+ 2 p)])
+
+(defn bst
+  [A B C oprand p]
+  [A
+   (mod (combo-oprand A B C oprand) 8)
+   C
+   nil
+   (+ 2 p)])
+
+(defn jnz
+  [A B C oprand p]
+  [A
+   B
+   C
+   nil
+   (if (zero? A) (+ 2 p) oprand)])
+
+(defn bxc
+  [A B C _ p]
+  [A
+   (bit-xor B C)
+   C
+   nil
+   (+ 2 p)])
+
+(defn out
+  [A B C oprand p]
+  [A
+   B
+   C
+   (mod (combo-oprand A B C oprand) 8)
+   (+ 2 p)])
+
+(defn bdv
+  [A B C oprand p]
+  [A
+   (quot A (int (math/pow 2 (combo-oprand A B C oprand))))
+   C
+   nil
+   (+ 2 p)])
+
+(defn cdv
+  [A B C oprand p]
+  [A
+   B
+   (quot A (int (math/pow 2 (combo-oprand A B C oprand))))
+   nil
+   (+ 2 p)])
+
+(def opcode-to-fn
+  {0 adv
+   1 bxl
+   2 bst
+   3 jnz
+   4 bxc
+   5 out
+   6 bdv
+   7 cdv })
+
+(defn d17p1
+  [input]
+  (let [[A B C instr] (parse-input input) ]
+    (loop [A A
+           B B
+           C C
+           p 0
+           outs []]
+      (let [opcode (get instr p)
+            oprand (get instr (inc p))]
+        (if (nil? opcode)
+          (->> outs
+               (map (comp str int))
+               (interpose ",")
+               (apply str))
+          (let [[A B C out p] ((get opcode-to-fn opcode) A B C oprand p)
+;;                _ (prn [:operation opcode oprand])
+;;                _ (prn [A B C out p])
+                outs (if out (conj outs out) outs)]
+            (recur A B C p outs)))))))
+
+(defn one-series
+  [A]
+  (let [B (mod A 8)
+        C (quot A (int (math/pow 2 (* 2 B))))
+        A (quot A (int (math/pow 2 (+ 3 B))))
+        B (bit-xor B 6)
+        B (bit-xor B C)]
+    [A out]
+    ))
+
+(defn d17p2-brute
+  [input]
+  (let [[A B-i C-i instr] (parse-input input) ]
+    (loop [Astart 77500
+           A 0
+           B B-i
+           C C-i
+           p 0
+           outs []]
+      (let [opcode (get instr p)
+            oprand (get instr (inc p))]
+        (if (nil? opcode)
+          (if (= outs instr)
+            Astart
+            (do (when (zero? (rem Astart 1000)) (prn [Astart outs instr]))
+                (recur (inc Astart) (inc Astart) B-i C-i 0 [])))
+          (let [[A B C out p] ((get opcode-to-fn opcode) A B C oprand p)
+                outs (if out (conj outs out) outs)]
+            (comment
+              (prn [:start Astart])
+              (prn [:operation opcode oprand])
+              (prn [A B C out p])
+              (prn [:instr instr :outs outs])
+              )
+            (if (every? true? (map = outs instr))
```

Comment:
Is there a specific reason for this `every?`? As far as I know this should be semantically equivalent:
```suggestion
            (if (= outs instr)
```

Reply:
yes, because I expect `outs` to be smaller than `instr` so they are not equals but what exists of the elements are. So basically, if the first output is wrong, I don't bother with the rest of the program and test the next A (or the second ...)
That was in hope that I could pass the "obviously wrong" A faster and reach an answer (before checking by hand the expected range of A)